dashboard/test/controllers/omniauth_callbacks_controller_test.rb

```diff
@@ -416,6 +416,50 @@ class OmniauthCallbacksControllerTest < ActionController::TestCase
     end
   end
 
+  test 'login: oauth takeover takes over account when account has no activity' do
+    User::OAUTH_PROVIDERS_UNTRUSTED_EMAIL.each do |provider|
+      oauth_student = create :student, provider: provider, uid: '12345'
+      student = create :student
+
+      @request.session[ACCT_TAKEOVER_EXPIRATION] = 5.minutes.from_now
+      @request.session[ACCT_TAKEOVER_PROVIDER] = provider
+      @request.session[ACCT_TAKEOVER_UID] = oauth_student.uid
+      @request.session[ACCT_TAKEOVER_OAUTH_TOKEN] = '54321'
```

Comment:
Should we extract a `begin_fake_takeover` helper to deduplicate between these two tests?

Reply:
:+1: